File: api/PetitionPower/PetitionPower.Web/Services/Common/Dtos/PagedResultDto.cs
namespace PetitionPower.Web.Services.Common.Dtos;

public record PagedResultDto<TEntityDto>
{
    public PagedResultDto(int page, int pageSize, int totalCount, List<TEntityDto> items)
    {
        Page = page;
        TotalCount = totalCount;

        TotalPages = (int)Math.Ceiling(totalCount / (double)pageSize);
        Items = items;
    }

    public int Page { get; set; }
    public int TotalCount { get; set; }
    public int TotalPages { get; set; }

    public bool HasNext => TotalPages > Page;
    public bool HasPrevious => Page > 1;

    public List<TEntityDto> Items { get; set; }
}

File: client_frontend/src/api/pagedResultDto.ts
export interface PagedResultDto<TEntityDto> {
    page: number;
    totalCount: number;
    totalPages: number;
    hasNext: boolean;
    hasPrevious: boolean;
    items: TEntityDto[];
}

export interface PagedRequestDto {
    page: number;
    pageSize: number;
    sorting?: string | null;
}